src/api/UsersApi.ts

```diff
@@ -35,9 +43,23 @@ export const updateUser = (
 ): AxiosPromise<UserT> =>
     API.put(compilePath(API_PATHS.USER, { key }), payload);
 
+export const approveUserAwaitingApproval = (
+    key: string,
+    payload: UserApprovalPayloadT
+): AxiosPromise<UserT> =>
+    API.put(
+        API_PATHS.USERS_AWAITING_APPROVAL.concat(`?username=`).concat(key),
+        payload
+    );
```

Comment:
According to the [axios documentation](https://axios-http.com/docs/req_config) I think you can replace this with something like


```suggestion
    API.put(
        API_PATHS.USERS_AWAITING_APPROVAL,
        payload,
        {"params": {"username": key}}
    );
```

(I haven't tried it)